log4j-core/src/main/java/org/apache/logging/log4j/core/config/LoggerConfig.java

```diff
@@ -614,15 +611,18 @@ public ReliabilityStrategy getReliabilityStrategy() {
      * Logs an event, bypassing filters.
      *
      * @param event     The log event.
-     * @param predicate predicate for which LoggerConfig instances to append to. A
-     *                  {@literal null} value is equivalent to a true predicate.
+     * @param dispatchType decides which logger configs should be used.
      */
-    protected void processLogEvent(final LogEvent event, final LoggerConfigPredicate predicate) {
+    protected void processLogEvent(final LogEvent event, final DispatchType dispatchType) {
         event.setIncludeLocation(isIncludeLocation());
-        if (predicate == null || predicate.allow(this)) {
+        if (hasDispatchType(dispatchType)) {
             callAppenders(event);
         }
-        logParent(event, predicate);
+        logParent(event, dispatchType);
+    }
+
+    protected boolean hasDispatchType(final DispatchType dispatchType) {
+        return dispatchType != DispatchType.ASYNCHRONOUS;
```

Comment:
I am not able to follow the logic here. `hasDispatchType()` sounds like we will check if there are any dispatch types. Though we certainly don't do that. Plus, we only check if it is not async; again something I couldn't understand. Mind elaborating on what are you trying to do here? I was expecting something like `boolean dispatchTypeAllowed(DispatchType)`.

Also, I am puzzled to see _async_ in `LoggerConfig`. `DispatchType` will still litter the `LoggerConfig` with `async` even if we split it into a separate module. Am I wrong? To avoid this, shouldn't `DispatchType` be a protected interface that integrators extend from instead?

Reply:
I replaced `LoggerConfigPredicate` with `Predicate<LoggerConfig>`.
All references to async in `log4j-core` should be gone.